tools/service_image_uploader/downloader.toit

```diff
@@ -80,32 +80,34 @@ download config/cli.Config cache/cli.Cache ui/Ui parsed/cli.Parsed:
     service-images.do: | row |
       image := row["image"]
       cache-key := "snapshot-downloader/$image"
-      snapshot := cache.get cache-key: | store/cli.FileStore |
+      cache.get cache-key: | store/cli.FileStore |
         ui.info "Downloading $row["sdk_version"]-$row["service_version"]."
+        snapshot/ByteArray? := null
         exception := catch:
-          store.save (client.storage.download --path="service-snapshots/$image")
+          snapshot = client.storage.download --path="service-snapshots/$image"
         if exception:
           ui.error "Failed to download $row["sdk_version"]-$row["service_version"]."
           ui.error "Are you logged in as an admin?"
           ui.error exception
           ui.abort
 
-      ar-reader := ar.ArReader (io.Reader snapshot)
-      artemis-header := ar-reader.find AR-SNAPSHOT-HEADER
-      if not artemis-header:
-        // Deprecated direct snapshot format.
-        uuid := cache-snapshot snapshot --output-directory=output-directory
-        ui.info "Wrote service snapshot $uuid."
-      else:
-        // Reset the reader.
-        // The header should be the first file, but it doesn't cost much to start
-        // from scratch and avoid hard-to-find bugs.
-        ar-reader = ar.ArReader (io.Reader snapshot)
-        while file/ar.ArFile? := ar-reader.next:
-          if file.name == AR-SNAPSHOT-HEADER:
-            continue
-          uuid := cache-snapshot file.content --output-directory=output-directory
+        ar-reader := ar.ArReader (io.Reader snapshot)
```

Comment:
This part is now intended.
This means that the downloader will not repopulate the snapshot-directory if the downloader-cache already contains the information. This can happen if one deletes the Jaguar/Toit cache but not the Artemis cache.
Easy fix: delete the downloader cache as well.

In return the downloader is now much faster for files that are already on disk.